crates/payload/validator/src/lib.rs

```diff
@@ -116,7 +116,7 @@ impl<ChainSpec: EthereumHardforks> ExecutionPayloadValidator<ChainSpec> {
         &self,
         payload: ExecutionPayload,
         sidecar: ExecutionPayloadSidecar,
-    ) -> Result<SealedBlock<Header, BlockBody<T>>, PayloadError> {
+    ) -> Result<SealedBlock<reth_primitives::Block<T>>, PayloadError> {
```

Comment:
should this method also just be parameterized by `B` instead of tx?

Reply:
yeah we can simplify this a lot, I ported everything so far but once ready we should do another simplification pass